### DISCUSSION TITLE
Querystring property/method produces deprecated warning

### DISCUSSION THREAD

**Discussion by areichardt-apo** [livewire/livewire#8057] [7. Bugs (archived)]
### Livewire version

v3.2.6

### Laravel version

v10.46.0

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

Create a component like this, using the queryString method or property:

```php
class Test extends Component {

  public $search = '';

  public function render(): View {
    return view('livewire.test');
  }

  protected function queryString(): array {
    return [
      'search'
    ];
  }
}
```

This will result in a warning no matter if you pass the query string parameter or not:

`ReflectionClass::hasProperty(): Passing null to parameter #1 ($name) of type string is deprecated in /var/www/vendor/livewire/livewire/src/Features/SupportQueryString/BaseUrl.php on line 42`

This is the line:

`if ($reflectionClass->hasProperty($this->getSubName())) {`

getSubName() is null here.

Its working if you use the #[Url] attribute for $search.

We have a lot of componentes using the Livewire 2 approach with the queryString property and it was working fine up to release v3.4.4.

Even if the queryString property is not mentioned anymore in the documentation, the issues happens also with the queryString method and this one is still part of the documentation.

### Repository or Wirebox URL

empty

### Additional information

No wirebox url as you do not see the Warning.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@areichardt-apo any chance you could submit a failing test PR so we can investigate? Thanks!

**Reply by areichardt-apo** [reply]
Done, is this like you wanted?

**Reply by areichardt-apo** [reply]
https://github.com/livewire/livewire/pull/8084

**Reply by joshhanley** [reply]
@areichardt-apo thanks!
